1.3 release notes (#3880)
elastic/cloud-on-k8s: docs/release-notes/highlights-1.3.0.asciidoc

[[release-highlights-1.3.0]]
== 1.3.0 release highlights

[float]
[id="{p}-130-new-and-notable"]
=== New and notable

New and notable changes in version 1.3.0 of {n}. See <<release-notes-1.3.0>> for the full list of changes.

[float]
[id="{p}-130-official-helm-chart"]
==== Official Helm chart

> Addressing one of the most common asks, ECK 1.3.0 introduces <<{p}-install-helm,an official Helm chart>> which allows to configure and deploy ECK. Apart from enabling new deployments, a migration path that enables moving from previous deployment method to a Helm chart is documented.

Reviewer: ```suggestion
Addressing one of the most common asks, ECK 1.3.0 introduces <<{p}-install-helm,an official Helm chart>> which allows to configure and deploy ECK. The documentation describes how to enable new deployments, and how to migrate from previous deployment methods to Helm chart.
```

Reviewer 2: ```suggestion
ECK 1.3.0 introduces <<{p}-install-helm,an official Helm chart>>. Helm is a popular package manager for Kubernetes. Apart from enabling new deployments, a migration path that enables moving from previous deployment method to a Helm chart is documented.
```

> Addressing one of the most common asks, ECK 1.3.0 introduces <<{p}-install-helm,an official Helm chart>> which allows to configure and deploy ECK. Apart from enabling new deployments, a migration path that enables moving from previous deployment method to a Helm chart is documented.

Reviewer 3: ECK 1.3.0 introduces an official Helm chart for deploying the operator. This provides a more flexible way of installing the operator customized to the environment and the desired use case. Existing installations can be easily migrated to Helm as well.

[float]
[id="{p}-130-volume-expansion-support"]
==== Volume expansion support

> Elasticsearch Pods volumes can be now dynamically expanded. A single edit to the resource YAML manifest is all that's needed to grow the volumes. In addition to that, in many cases (including GCE-PD, AWS-EBS, Azure, and TopoLVM), it is a hot filesystem resize that doesn't require Pod restart, and it's active seconds after the change was made.

Reviewer: ```suggestion
Elasticsearch Pods volumes can be now dynamically expanded. A single edit to the resource YAML manifest is enough to grow the volumes. In addition to that, in many cases (including GCE-PD, AWS-EBS, Azure, and TopoLVM), it is a hot filesystem resize that doesn't require Pod restart, and it's active seconds after the change was made.
```

Reviewer 2: ```suggestion
Elasticsearch volumes can be now dynamically expanded. A single edit to the resource YAML manifest is enough to grow the volumes. In many cases (dependent on the underlying storage provider), this can be done without even restarting the pod and so expansions complete in seconds.
```

> Elasticsearch Pods volumes can be now dynamically expanded. A single edit to the resource YAML manifest is all that's needed to grow the volumes. In addition to that, in many cases (including GCE-PD, AWS-EBS, Azure, and TopoLVM), it is a hot filesystem resize that doesn't require Pod restart, and it's active seconds after the change was made.

Reviewer 3: ```suggestion
Elasticsearch volumes can now be dynamically expanded. A single edit to the resource YAML manifest is enough to grow the volumes. In many cases, if the storage provider supports it, this will take effect almost immediately without even requiring a pod restart.
```

[float]
[id="{p}-130-shard-allocation-is-node-aware-by-default"]
==== Shard allocation is node aware by default

> ECK-managed Elasticsearch can be configured to be node aware - Elasticsearch will recognize that shard replicas can't be allocated to the same Kubernetes node and instead, spread them over multiple nodes. This prevents a single node issue affecting the entire shard. Until now, achieving this behavior required a proper configuration from the user. With ECK 1.3.0, the configuration enabling this is the default.

Reviewer: ```suggestion
ECK-managed Elasticsearch can be configured to be node aware - Elasticsearch will recognize that shard replicas can't be allocated to the same Kubernetes node and instead, spread them over multiple nodes. This prevents a single node issue affecting the entire shard. Until now, this behavior required a proper configuration from the user. With ECK 1.3.0, this is part of the default configuration.
```

> ECK-managed Elasticsearch can be configured to be node aware - Elasticsearch will recognize that shard replicas can't be allocated to the same Kubernetes node and instead, spread them over multiple nodes. This prevents a single node issue affecting the entire shard. Until now, achieving this behavior required a proper configuration from the user. With ECK 1.3.0, the configuration enabling this is the default.

Reviewer 2: ```suggestion
Elasticsearch can be configured to be aware of the underlying Kubernetes node it is running on, and use this knowledge when deciding where to place replica shards. This prevents an issue with a single Kubernetes node from affecting both primary and replica shards. In previous ECK releases, this behavior required additional configuration by the user. With ECK 1.3.0, this configuration is the default.
```

[float]
[id="{p}-130-ipv6-compatibility"]
==== IPv6 compatibility

> With changes around how IP addresses are handled, ECK is now compatible with IPv6 enabled environments.

Reviewer 2: This could use more explanation I think.

[float]
[id="{p}-130-ubi-base-image"]
==== UBI base image

> To enable ECK becoming link:https://connect.redhat.com/en/partner-with-us/red-hat-openshift-operator-certification[Red Hat OpenShift Certified Operator], ECK image is now based on Universal Base Image - `ubi-minimal`.

Reviewer 2: Is this how we decided to message it? If so I feel like we should have more detail around the benefits of being certified.

Reviewer 4: I am not sure how much in detail we want to go into the benefits of certification, but from the perspective of customers, I would add:  For customers using ECK on Openshift, we now have a certified operator on OperatorHub

[float]
[id="{p}-130-init-containers-modifications"]
==== Init containers modifications

> All supported resources can now have their init containers modified.

Reviewer 2: Same here I think this deserves some more explanation, or maybe not including. You could always modify init containers, it was just a straight replacement :)

Author: I thought modified implies changes to the existing ones not replacement, but yeah, reader might lack context to get the difference right away. I'll add a bit more. Also, moving this to breaking changes.
